src/node_internals.h

```diff
@@ -203,6 +297,14 @@ void SetupProcessObject(Environment* env,
                         int exec_argc,
                         const char* const* exec_argv);
 
+/**
+ * Call _register<module_name> functions for all of
+ * the built-in modules. Because built-in modules don't
+ * use the __attribute__((constructor)). Need to
+ * explicitly call the _register* functions.
+ */
+void RegisterBuiltinModules();
```

Comment:
Sorry, I was unclear, but this is part of the embedder API so it should be in the public header – right?

Reply:
Hi @addaleax and @gireeshpunathil,  actually, this change is for built-in modules. It only removes the `__attribute__((constructos))` for build-in modules. So, I moved this to the `node_internals.h` and it needs to be invoked inside the `node::Start()`. 

For third-party modules, they are built as shared library in most case and `__attribute__((constructor))` works properly in `dlopen()`. I left that part as-is. I was not aware that xIc compiler doesn't support it. If we want to remove the `__attribute__((constructor))` for third-party modules, we also need to change the module initialization process.

Reply:
@yhwang Yes, but not all embedders use `node::Start(int argc, char** argv)`, right? So they need a definition of this function to be available.